lib/matplotlib/tight_layout.py

```diff
@@ -68,8 +68,17 @@ def auto_adjust_subplotpars(
 
     if ax_bbox_list is None:
         ax_bbox_list = [
-            Bbox.union([ax.get_position(original=True) for ax in subplots])
-            for subplots in subplot_list]
+            [ax.get_position(original=True) for ax in subplots]
+            for subplots in subplot_list
+        ]
+        ax_bbox_list = [
+            Bbox.union([
+                b for b in bbox_list
+                if np.isfinite(b.width) and np.isfinite(b.height)
+                    and (b.width != 0 or b.height != 0)
```

Comment:
No tests fail if this isn't added, but it needs to be here now that I return `Bbox.null`, since a `Union` that includes `Bbox.null` will always resolve to the whole space (a separate issue, which should be addressed once I wrap up #17115).

Note that `Figure.get_tightbbox` already does the same thing.